### PR TITLE
Remove support for MySQL v5

### LINKED ISSUE #1338: Deprecation Notice: MySQL v5 support
MySQL v5 goes EOL in 5 months from now: https://endoflife.date/mysql

This is a notice that we will soon deprecate v5 support and move the MySQL datastore integration tests to use v8.

### PR DESCRIPTION
**Pull request by josephschorr**
Fixes #1338

To be merged no earlier than Oct 31, when MySQL v5 reaches EOL

### CODE REVIEW

File: internal/datastore/mysql/datastore_test.go [hunk 1]
```diff
@@ -94,13 +94,6 @@ func TestMySQLDatastoreDSNWithoutParseTime(t *testing.T) {
...
 }
 
-func TestMySQL5Datastore(t *testing.T) {
-	b := testdatastore.RunMySQLForTesting(t, "")
-	dst := datastoreTester{b: b, t: t}
-	test.All(t, test.DatastoreTesterFunc(dst.createDatastore))
-	additionalMySQLTests(t, b)
-}
-
 func TestMySQL8Datastore(t *testing.T) {
```
Comment (ecordell) [inline internal/datastore/mysql/datastore_test.go:97]: https://github.com/josephschorr/spicedb/blob/608f9eccb7ec17b63882f985f5ee90fb8f6d5649/internal/testserver/datastore/mysql.go#L37

I think you want to remove a bunch of the v5 stuff here as well

Comment (josephschorr) [reply internal/datastore/mysql/datastore_test.go:97]: Looks like the file forgot to save; VSCode does that occasionally

**Comment by vroldanbet** [comment]
>To be merged no earlier than Oct 31, when MySQL v5 reaches EOL

We can probably merge it already as the next release will almost certainly happen after Oct 31st

**Review by vroldanbet — APPROVED** [review]
LGTM

**Pull request merged**